lib/active_model/serializer/adapter/attributes.rb

```diff
@@ -51,10 +23,48 @@ def fragment_cache(cached_hash, non_cached_hash)
 
         private
 
+        def serializable_hash_for_collection(options)
+          serializer.map { |s| Attributes.new(s, instance_options).serializable_hash(options) }
+        end
+
+        def serializable_hash_for_single_resource(options)
+          resource = resource_object_for(options)
+          relationships = resource_relationships(options)
+          resource.merge!(relationships)
+        end
+
+        def resource_relationships(options)
+          relationships = {}
+          serializer.associations(@include_tree).each do |association|
+            relationships[association.key] = relationship_value_for(association, options)
+          end
+
+          relationships
+        end
+
+        def relationship_value_for(association, options)
```

Comment:
I guess if this isn't the kind of method that is ever going to be overridden when it moves to Adapter::Base, I can get behind this. It at least looks a little better